resources/migrations/000_migrations.yaml

```diff
@@ -11845,7 +11845,51 @@ databaseChangeLog:
             defaultNullValue: '[]'
             tableName: report_card
 
-
+  - changeSet:
+      id: v44.00-033
+      author: qnkhuat
+      comment: >-
+        Added 0.43.0. Grant the 'All Users' Permissions Group readwrite perms for the Root Snippets Collection.
+      preConditions:
+        - onFail: MARK_RAN
+        # HACK: to prevent this migration from running on existing EE instances (#21940)
+        - sqlCheck:
+            expectedResult: 0
+            sql: >-
+              SELECT CASE
+                        /* Run for any new instances */
+                        WHEN user_count = 0 THEN 0
```

Comment:
Wouldn't this be enough, since it's not possible to have 0 users, while having a token setting? Also, many setups uses envvars to define token, so there won't be a setting in the appdb.
It would make it easier to read the migration - less is more.

Reply:
> Also, many setups use envvars to define tokens, so there won't be a setting in the appdb.

My original thinking is that we should run this migration for existing OSS instances so that if they update to EE, this change will be included. 

But if we can't know for sure if an instance is OSS or EE by just reading the DB, I'll make this migration run on new instances only.